src/views/signup.jsx

```diff
@@ -1,7 +1,185 @@
-import React from "react";
+import { useState } from "react";
+import { Link, useNavigate } from "react-router-dom";
+import {
+  createUserWithEmailAndPassword,
+  GoogleAuthProvider,
+  signInWithPopup,
+} from "firebase/auth";
+import { auth } from "../../firebase_config";
 
-const signup = () => {
-  return <div>signup</div>;
-};
+function Signup() {
+  const [email, setEmail] = useState("");
+  const [password, setPassword] = useState("");
+  const [confirmPassword, setConfirmPassword] = useState("");
 
-export default signup;
+  const provider = new GoogleAuthProvider();
+  const navigate = useNavigate();
+
+  const handleSignUp = async (e) => {
+    e.preventDefault();
+
+    if (password !== confirmPassword) {
+      alert("password don't match");
+      return;
+    }
+
+    createUserWithEmailAndPassword(auth, email, password)
+      .then(() => {
+        alert("sign up successfully");
```

Comment:
we may want to consider using a more user-friendly way to display errors, such as a toast notification or an inline error messages. I think we talked about using the npm package toasty for this.